core/commands/mount_unix.go

```diff
@@ -73,74 +77,53 @@ baz
 `,
 	},
 	Options: []cmdkit.Option{
-		cmdkit.StringOption("ipfs-path", "f", "The path where IPFS should be mounted."),
-		cmdkit.StringOption("ipns-path", "n", "The path where IPNS should be mounted."),
+		cmdkit.StringOption(mountIPFSPathOptionName, "f", "The path where IPFS should be mounted."),
+		cmdkit.StringOption(mountIPNSPathOptionName, "n", "The path where IPNS should be mounted."),
 	},
-	Run: func(req cmds.Request, res cmds.Response) {
-		cfg, err := req.InvocContext().GetConfig()
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		cfg, err := cmdenv.GetConfig(env)
 		if err != nil {
-			res.SetError(err, cmdkit.ErrNormal)
-			return
+			return err
 		}
 
-		node, err := req.InvocContext().GetNode()
+		nd, err := cmdenv.GetNode(env)
 		if err != nil {
-			res.SetError(err, cmdkit.ErrNormal)
-			return
+			return err
 		}
 
 		// error if we aren't running node in online mode
-		if node.LocalMode() {
-			res.SetError(ErrNotOnline, cmdkit.ErrClient)
-			return
+		if nd.LocalMode() {
+			return err
 		}
 
-		fsdir, found, err := req.Option("f").String()
-		if err != nil {
-			res.SetError(err, cmdkit.ErrNormal)
-			return
-		}
+		fsdir, found := req.Options[mountIPFSPathOptionName].(string)
 		if !found {
 			fsdir = cfg.Mounts.IPFS // use default value
 		}
 
 		// get default mount points
-		nsdir, found, err := req.Option("n").String()
-		if err != nil {
-			res.SetError(err, cmdkit.ErrNormal)
-			return
-		}
+		nsdir, found := req.Options[mountIPNSPathOptionName].(string)
 		if !found {
 			nsdir = cfg.Mounts.IPNS // NB: be sure to not redeclare!
 		}
 
-		err = nodeMount.Mount(node, fsdir, nsdir)
+		err = nodeMount.Mount(nd, fsdir, nsdir)
 		if err != nil {
-			res.SetError(err, cmdkit.ErrNormal)
-			return
+			return err
 		}
 
 		var output config.Mounts
 		output.IPFS = fsdir
 		output.IPNS = nsdir
-		res.SetOutput(&output)
+		return cmds.EmitOnce(res, &output)
 	},
 	Type: config.Mounts{},
-	Marshalers: cmds.MarshalerMap{
-		cmds.Text: func(res cmds.Response) (io.Reader, error) {
-			v, err := unwrapOutput(res.Output())
-			if err != nil {
-				return nil, err
-			}
-
-			mnts, ok := v.(*config.Mounts)
-			if !ok {
-				return nil, e.TypeErr(mnts, v)
-			}
-
-			s := fmt.Sprintf("IPFS mounted at: %s\n", mnts.IPFS)
-			s += fmt.Sprintf("IPNS mounted at: %s\n", mnts.IPNS)
-			return strings.NewReader(s), nil
-		},
+	Encoders: cmds.EncoderMap{
+		cmds.Text: cmds.MakeTypedEncoder(func(req *cmds.Request, w io.Writer, mounts *config.Mounts) error {
+			s := fmt.Sprintf("IPFS mounted at: %s\n", mounts.IPFS)
```

Comment:
Let's just call `fmt.Fprintf(w, "...", ...)` twice. No need to build a string up first.